src/main/resources/api/service_openapi.yaml

```diff
@@ -191,6 +194,228 @@ paths:
         '500':
           $ref: '#/components/responses/ServerError'
 
+  /api/workspaces/v1/{id}/resources/references/datarepo/snapshot:
+    parameters:
+    - $ref: '#/components/parameters/Id'
+    post:
+      summary: Create a new data repo snapshot reference in a workspace.
+      operationId: createDataRepoSnapshotReference
+      tags: [DataRepoReference]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateDataRepoSnapshotReferenceRequestBody'
+      responses:
+        '200':
+          $ref: '#/components/responses/DataRepoSnapshotReferenceResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+  /api/workspaces/v1/{id}/resources/references/datarepo/snapshot/{referenceId}:
+    parameters:
+    - $ref: '#/components/parameters/Id'
+    - $ref: '#/components/parameters/ReferenceId'
+    get:
+      summary: Gets a reference to a snapshot from a workspace.
+      operationId: getDataRepoSnapshotReference
+      tags: [DataRepoReference]
+      responses:
+        '200':
+          $ref: '#/components/responses/DataRepoSnapshotReferenceResponse'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+    delete:
+      summary: Deletes a reference to a data repo snapshot from a workspace.
+      operationId: deleteDataRepoSnapshotReference
+      tags: [DataRepoReference]
+      responses:
+        '204':
+          description: OK
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+  /api/workspaces/v1/{id}/resources/references/datarepo/snapshot/name/{name}:
```

Comment:
Is name URL safe? Or you just assume it is encoded?

Reply:
Names must be URL safe and they're checked for validity in Controller methods